module/zfs/dbuf.c

```diff
@@ -400,7 +403,7 @@ dbuf_hash_insert(dmu_buf_impl_t *db)
 	uint32_t i;
 
 	blkid = db->db_blkid;
-	hv = dbuf_hash(os, obj, level, blkid);
+	hv = db->db_hv;
```

Comment:
maybe assert that the dbuf_hv still matches dbuf_hash(...)?

Reply:
Done.